Spec: prohibit zero opcodes (#801)
foxglove/mcap: docs/specification/README.md

## Records

> MCAP files may contain a variety of records. Records are identified by a single-byte **opcode**. Record opcodes in the range 0x01-0x7F are reserved for future MCAP format usage. 0x80-0xFF are reserved for application extensions and user proposals. The opcode 0x00 is an error.

Reviewer: Possibly clearer about what kind of error it is:

```suggestion
MCAP files may contain a variety of records. Records are identified by a single-byte **opcode**. Record opcodes in the range 0x01-0x7F are reserved for future MCAP format usage. 0x80-0xFF are reserved for application extensions and user proposals. 0x00 is not a valid opcode.
```

Reviewer 2: 👍
